core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -143,6 +143,10 @@ private TableProperties() {
   public static final String DELETE_PARQUET_PAGE_SIZE_BYTES = "write.delete.parquet.page-size-bytes";
   public static final int PARQUET_PAGE_SIZE_BYTES_DEFAULT = 1024 * 1024; // 1 MB
 
+  public static final String PARQUET_PAGE_ROW_COUNT_LIMIT = "write.parquet.page-row-count-limit";
+  public static final String DELETE_PARQUET_PAGE_ROW_COUNT_LIMIT = "write.delete.parquet.page-row-count-limit";
+  public static final int PARQUET_PAGE_ROW_COUNT_LIMIT_DEFAULT = 20_000;
```

Comment:
Is this the Parquet default? How did you decide on 20k?

Reply:
Yes, that's the Parquet default, so this shouldn't change any behavior if it isn't set.

Reply:
[Here](https://github.com/apache/parquet-mr/blob/0819356a9dafd2ca07c5eab68e2bffeddc3bd3d9/parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java#L58) is where the default it set: